addons/account/models/account_invoiceable_order.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class InvoiceableOrder(models.AbstractModel):
+    _name = 'account.invoiceable.order'
+    _description = 'Enables downpayment support for models that can have down payments on them'
+
+    amount_to_invoice = fields.Monetary("Amount to invoice", store=True, compute='_compute_amount_to_invoice')
+    amount_invoiced = fields.Monetary("Already invoiced", compute='_compute_amount_invoiced')
+
+    # To override
+    company_id = fields.Many2one('res.company')
+    currency_id = fields.Many2one('res.currency')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position')
+    partner_id = fields.Many2one('res.partner')
+
+    amount_total = fields.Monetary()
+
+    invoice_ids = fields.Many2many('account.move')
+    order_line = fields.One2many('account.invoiceable.line')
+
+    invoice_status = fields.Selection([])
+
+    def _qty_field_name(self):
+        raise NotImplementedError # To override
```

Comment:
It's a bit sad that we don't have the same names across different modules and need this kind of method :/
[ignore this]